src/Compilers/Test/Core/CompilationVerifier.cs

```diff
@@ -228,6 +233,110 @@ public void VerifyTypeIL(string typeName, string expected)
             }
         }
 
+        private sealed class Resolver : ILVerify.ResolverBase
+        {
+            private readonly Dictionary<string, ImmutableArray<byte>> _imagesByName = new Dictionary<string, ImmutableArray<byte>>(StringComparer.OrdinalIgnoreCase);
+
+            internal Resolver(IList<ModuleData> allModuleData)
+            {
+                foreach (var module in allModuleData)
+                {
+                    string name = module.SimpleName;
+                    if (_imagesByName.ContainsKey(name))
+                    {
+                        throw new Exception($"Multiple modules named '{name}' were found");
+                    }
+                    _imagesByName.Add(name, module.Image);
+                }
+            }
+
+            protected override PEReader ResolveCore(string simpleName)
+            {
+                if (_imagesByName.TryGetValue(simpleName, out var image))
+                {
+                    return new PEReader(image);
+                }
+
+                throw new Exception($"ILVerify was not able to resolve a module named '{simpleName}'");
+            }
+        }
+
+        private void ILVerify(Verification verification)
+        {
+            if (verification == Verification.Skipped)
+            {
+                return;
+            }
+
+            try
+            {
+                var resolver = new Resolver(_allModuleData);
+                var verifier = new ILVerify.Verifier(resolver);
+
+                var mscorlibModule = _allModuleData.Single(m => m.IsCorLib);
```

Comment:
>var mscorlibModule = _allModuleData.Single(m => m.IsCorLib); [](http://example.com/codeflow?start=16&length=60)

Consider moving outside of `try { } catch { }`. (It looks like this throws an `InvalidOperationException` from `Microsoft.CodeAnalysis.CSharp.UnitTests.UseSiteErrorTests.MissingTypeKindBasisTypes()`.